### ISSUE TITLE
[python] Reported evaluation metric and early stop

### ISSUE THREAD

**Issue by mlisovyi** [Microsoft/LightGBM#1318]
The issue that I face is: when one runs with the early stopping enabled, one aims to be able to **stop specifically on the eval_metric metric**. However, python API of LightGBM checks __all__ metrics that are monitored. This is different from the XGBoost choice, where they check the last item from the eval list, but this is also a justifiable choice. However, this choice should not prevent us from reaching the aforementioned goal.

There is some undocumented feature of the code, related to the list of evaluation metrics that are considered. Namely, if one uses any predefined metric as the objective (with the corresponding name string), then the objective metric is prepended to the list of evaluation metrics. Which is unexpected on the user side and not desired in terms of the goal defined before.  If one uses a custom objective function, then there no such extra metric prepended. 

Is there a chance to either have the early-stopping criterion to be configurable (with an extra argument in the fit function) or to drop the objective that appears in the evaluation result and that is not documented anywhere?

### Environment
I observe this feature with lightgbm v2.1.0.

### How to reproduce
Here is a   basic toy example:
```python
import numpy as np
import lightgbm as lgb

# generate random data
X = np.random.randn(10000,5)
y = np.random.randn(10000)

# define MSE evaluation and objective functions (for the sake of example)
def evaluate_mse(truth, prediction):
    return('xxx', np.mean(np.square(truth - prediction)), False)

def obj_mse(truth, prediction):
    grad = 2 * (prediction - truth)
    hess = 2 * np.ones(shape=truth.shape)
    return(grad, hess)

clf = lgb.LGBMRegressor()
# Train with the default 'mse' objective
clf.set_params(**{'objective': 'mse'})
clf.fit(X, y, 
        eval_set=[(X,y)],
        eval_names=['validation_0'],
        eval_metric=evaluate_mse,
        verbose=50)
# Train with the custom implementation of the MSE
clf.set_params(**{'objective': obj_mse})
clf.fit(X, y, 
        eval_set=[(X,y)],
        eval_names=['validation_0'],
        eval_metric=evaluate_mse,
        verbose=50)
```
In the example above there is no early stopping, but you can already see the mismatch in the monitored metrics

**Comment by guolinke** [comment]
you can try the latest master branch with following code:
```
import numpy as np
import lightgbm as lgb

# generate random data
X = np.random.randn(10000,5)
y = np.random.randn(10000)

# define MSE evaluation and objective functions (for the sake of example)
def evaluate_mse(truth, prediction):
    return('xxx', np.mean(np.square(truth - prediction)), False)

def obj_mse(truth, prediction):
    grad = 2 * (prediction - truth)
    hess = 2 * np.ones(shape=truth.shape)
    return(grad, hess)

clf = lgb.LGBMRegressor()
# Train with the default 'mse' objective
clf.set_params(**{'objective': 'mse', 'metric': 'None'})
clf.fit(X, y, 
        eval_set=[(X,y)],
        eval_names=['validation_0'],
        eval_metric=evaluate_mse,
        verbose=50)
```

**Comment by guolinke** [comment]
@mlisovyi It is in the Parameter document. 
You can set it to `None', 'null' or 'na'. However, Python `None` equals to empty string, which will use the objective name as the metric name.

**Comment by mlisovyi** [comment]
Thanks a lot for pointing this out. There seems to be some room for improvement of the documentation: the section that you pointed out does say **nothing** about how the `metric` arg is used (only what are the possible values and even omitting the disabling values). Also in other places in the python API documentation (I can not say anything about docs of other API's) there is no mention that the `metric` parameter is also used to evaluate performance.

What is the policy on contributions to documentation updates? I'm fine to make a PR with text add-on's if you have no time for it.

**Comment by guolinke** [comment]
@mlisovyi it will be great if you can contribute on it.
You can just create a PR to fix it
